### PR TITLE
AP_Common: Fix AP_FWVersion after structure change

### PR DESCRIPTION
**Pull request by patrickelectric**
The change happened on 8814486572e9ad2f1611711f14604a0f9bca2b90, that breaks the ABI of the interface. 

output:
```
FWVersion:                                                                                                                                                                                                                                                                       
    header:                                                                                                                                                                                                                                                                      
        magic: apfwver                                                                                                                                                                                                                                                           
        checksum: 0xfb                                                                                                                                                                                                                                                           
        version: 2.0                                                                                                                                                                                                                                                             
        pointer_size: 4                                                                                                                                                                                                                                                          
    firmware:                                                                                                                                                                                                                                                                    
        string: ArduSub V4.2.0dev (16c5e640)                                                                                                                                                                                                                                     
        vehicle: ArduSub                                                                                                                                                                                                                                                         
        board: LINUX                                                                                                                                                                                                                                                             
        board subtype: LINUX_NAVIGATOR                                                                                                                                                                                                                                           
        hash: 16c5e640                                                                                                                                                                                                                                                           
        hash integer: 0x16c5e640                                                                                                                                                                                                                                                 
        version: 4.2.0                                                                                                                                                                                                                                                           
        type: Dev                                                                                                                                                                                                                                                                
    os:                                                                                                                                                                                                                                                                          
        name:                                                                                                                                                                                                                                                                    
        hash:                                                                                                                                                                                                                                                                    
        software_version: 0/0/0                                                                                                                                                                                                                                                  
    middleware:                                                                                                                                                                                                                                                                  
        name:                                                                                                                                                                                                                                                                    
        hash: ArduSub V4.2.0dev                                                                                                                                                                                                                                                  
```

### CODE REVIEW

**Review by Williangalvani — CHANGES_REQUESTED** [review]
works with firmwares before and after https://github.com/ArduPilot/ardupilot/commit/8814486572e9ad2f1611711f14604a0f9bca2b90, but I found a typo!

File: Tools/scripts/firmware_version_decoder.py [hunk 1]
```diff
@@ -192,6 +196,9 @@ def unpack_fwversion(self) -> None:
 
         self.fwversion.firmware_string = self.unpack_string_from_pointer()
         self.fwversion.firmware_hash_string = self.unpack_string_from_pointer()
+        if major_version >= 2:
+            self.firmware_hash = self.unpack("I")
```
[the remark this answers is not in the document]
Comment (patrickelectric) [reply Tools/scripts/firmware_version_decoder.py]: Thanks, fixed!

**Review by Williangalvani — APPROVED** [review]
lgtm, but let's wait for @peterbarker 's review

**Review by peterbarker — APPROVED** [review]

**Comment by peterbarker** [comment]
Merged, thanks!

... especially thanks for the comment to remove that little landmine :-)

**Pull request merged**